src/main/kotlin/org/jetbrains/intellij/tasks/RunIdeBase.kt

```diff
@@ -145,18 +124,22 @@ abstract class RunIdeBase(runAlways: Boolean) : JavaExec() {
         }
 
         if (!systemProperties.containsKey("idea.platform.prefix")) {
-            val matcher = VERSION_PATTERN.matcher(ideBuildNumber(ideDir.get().asFile))
-            if (matcher.find()) {
-                val abbreviation = matcher.group(1)
-                val prefix = prefixes[abbreviation]
-                if (!prefix.isNullOrEmpty()) {
-                    systemProperty("idea.platform.prefix", prefix)
-                }
+            info(context, "Looking for platform prefix")
+            val prefix = Files.list(ideDir.get().asFile.toPath().resolve("bin"))
+                    .asSequence()
+                    .filter { file -> file.fileName.toString().endsWith(".sh") }
+                    .flatMap { file -> Files.lines(file).asSequence() }
+                    .mapNotNull { line -> platformPrefixSystemPropertyRegex.find(line)?.groupValues?.getOrNull(1) }
+                    .firstOrNull()
+
+            if (prefix != null) {
+                systemProperty("idea.platform.prefix", prefix)
             }
+            info(context, "Using idea.platform.prefix=$prefix")
```

Comment:
IMHO you need to assert the case where prefix was not found by both system property and script search

Reply:
> where prefix was not found by both system property and script search

Unfortunately, it means that empty prefix should be used for IntelliJ Ultimate

Reply:
Yes, in that case, you may check that it's really IntelliJ Ultimate and not some other IDE

Reply:
makes sense. done